hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanActionExecutor.java

```diff
@@ -226,17 +220,13 @@ private HoodieCleanMetadata runClean(HoodieTable<T, I, K, O> table, HoodieInstan
           cleanStats,
           cleanerPlan.getExtraMetadata()
       );
-      if (!skipLocking) {
-        this.txnManager.beginTransaction(Option.of(inflightInstant), Option.empty());
-      }
+      this.txnManager.beginTransaction(Option.of(inflightInstant), Option.empty());
```

Comment:
hmm, since the outer lock was removed for clean table service, so it makes sense we always add a lock here.